packages/yoastseo/src/languageProcessing/helpers/word/getWordsForHTMLParser.js

```diff
@@ -2,18 +2,18 @@ import { punctuationRegexEnd, punctuationRegexStart } from "../sanitize/removePu
 import { hashedHtmlEntitiesRegexEnd, hashedHtmlEntitiesRegexStart } from "../../../helpers/htmlEntities";
 
 /*
- * The following regex matches a word separator. A word separator is either a whitespace, a slash, a backslash, a
- * tab or a non-breaking space.
+ * The following regex matches a word separator. A word separator is either a whitespace, a slash, a
+ * tab or a non-breaking space. Brackets are added to deal correctly with shortcodes downstream.
  * The regex is used to split a text into tokens.
  * Do not add punctuation marks to this regex, as they are handled separately inside splitIntoTokens().
  * The word separator explicitly only contains characters that split two words and not a word and a space.
- * A space is a word separator because it separates two words if it occurs between two words. For example: "foo bar"
- * A slash is a word separator because it separates two words if it directly borders those words. For example: "foo/bar"
- * A backslash is a word separator because it separates two words if it occurs between two words. For example: "foo\bar"
- * A tab is a word separator because it separates two words if it occurs between two words. For example: "foo	bar"
- * A non-breaking space is a word separator because it separates two words if it occurs between two words. For example: "foo\u00A0bar"
+ * - A space is a word separator because it separates two words if it occurs between two words. For example: "foo bar"
+ * - A tab is a word separator because it separates two words if it occurs between two words. For example: "foo	bar"
+ * - A non-breaking space is a word separator because it separates two words if it occurs between two words. For example: "foo\u00A0bar"
+ * - A slash is a word separator because it separates two words if it directly borders those words. For example: "foo/bar"
+ * - Open and closing brackets are added to deal correctly with shortcodes downstream.
  */
-const wordSeparatorsRegex = /([\s\t\u00A0])/;
+const wordSeparatorsRegex = /([\s\t\u00A0/[\]])/;
```

Comment:
Here, I added `[` and `]` to split shortcodes (and other text between brackets) from words, e.g. `[caption]test[/caption]`, or `test[1]`. 

I added the `/` as well, because that was there is the comment, but NOT in the code. The `\` gave problems downstream, so I'm not adding that here.

Reply:
According to this PR that introduces the `wordSeparatorsRegex`: https://github.com/Yoast/wordpress-seo/pull/20384 , the word boundaries are a space, a tab and a non-breaking-space. So the comment for the slash was on oversight I guess. Since it looks like it was a conscious decision not to include slash as word separator